playbooks/tasks/portal-role-task-mongo-replicaset-ensure-member.yml

```diff
@@ -0,0 +1,81 @@
+---
+
+# Ensure MongoDB node is member of the MongoDB replicaset
+# TODO: This ansible playbook task, should be moved to portal role
+
+# Notes:
+# - MongoDB primary node tasks should be executed on {{ mongo_primary_host }},
+#   not on the current Ansible host that we are handling.
+
+- name: Include getting MongoDB replicaset primary host
+  include_tasks: tasks/portal-role-task-mongo-replicaset-get-primary.yml
+
+- name: Check if MongoDB node is part of the replicaset
+  community.mongodb.mongodb_shell:
+    # Use secure localhost connection unless on subcrap test cluster
+    login_host: "{{ 'localhost' if (hostvars[mongo_primary_host].subcrap_mongo_port is not defined) else hostvars[mongo_primary_host].ansible_host }}"
+    # Use default port unless on subcrap test cluster
+    login_port: "{{ hostvars[mongo_primary_host].subcrap_mongo_port | default('27017') }}"
+    # Use delegated mongo_primary_host's mongo user, because we delegate this task
+    login_user: "{{ skynet_db_user_result.stdout }}"
+    # Use delegated mongo_primary_host's mongo password, because we delegate this task
+    login_password: "{{ skynet_db_pass_result.stdout }}"
+    mongo_cmd: "docker exec mongo mongo"
+    eval: |
+      rs.config().members.forEach(m => {
+        if (m.host == '{{ ansible_host }}:{{ subcrap_mongo_port | default('27017') }}')
+        {
+          print('is-member')
+        }
+      })
+    # transformed_output can't be parsed correctly by this module, keep it raw
+    transform: raw
+  register: mongo_primary_result
+  delegate_to: "{{ mongo_primary_host }}"
+  vars:
+    ansible_port: "{{ hostvars[mongo_primary_host].ansible_port | default(22) }}"
```

Comment:
is this port 22 because it is the ssh port for connecting to the primary node? If so, might be worth extending the variable name to something like `primary_node_ansible_ssh_port`